modules/javafx.graphics/src/main/native-glass/mac/PlatformSupport.m

```diff
@@ -0,0 +1,281 @@
+    [PlatformSupport putColor:preferences key:"macOS.NSColor.systemYellowColor" value:[NSColor systemYellowColor]];
+}
+
++ (void)updatePreferences:(jobject)application {
+    GET_MAIN_JENV;
+
+    jobject newPreferences = [self collectPreferences];
+    if (newPreferences == nil) {
+        return;
+    }
+
+    jboolean preferencesChanged = !(*env)->CallBooleanMethod(
+        env, newPreferences, jObjectEqualsMethod, currentPreferences);
+    GLASS_CHECK_EXCEPTION(env);
+
+    if (preferencesChanged) {
+        if (currentPreferences != nil) {
+            (*env)->DeleteGlobalRef(env, currentPreferences);
+        }
+
+        currentPreferences = (*env)->NewGlobalRef(env, newPreferences);
+
+        jobject unmodifiablePreferences = (*env)->CallStaticObjectMethod(
+            env, jCollectionsClass, jCollectionsUnmodifiableMapMethod, newPreferences);
+        GLASS_CHECK_EXCEPTION(env);
+
+        if (unmodifiablePreferences != nil) {
+            (*env)->CallVoidMethod(
+                env, application,
+                javaIDs.MacApplication.notifyPreferencesChanged,
+                unmodifiablePreferences);
+
+            (*env)->DeleteLocalRef(env, unmodifiablePreferences);
+        }
+
+    }
+
+    (*env)->DeleteLocalRef(env, newPreferences);
+}
+
++ (void)putString:(jobject)preferences key:(const char*)key value:(const char*)value {
+    GET_MAIN_JENV;
+
+    (*env)->CallObjectMethod(env, preferences, jMapPutMethod,
+        (*env)->NewStringUTF(env, key),
+        value != nil ? (*env)->NewStringUTF(env, value) : nil);
+}
+
++ (void)putColor:(jobject)preferences key:(const char*)colorName value:(NSColor*)color {
+    GET_MAIN_JENV;
+
+    NSColor* c = [color colorUsingColorSpace:[NSColorSpace deviceRGBColorSpace]];
+    (*env)->CallObjectMethod(env, preferences, jMapPutMethod,
+        (*env)->NewStringUTF(env, colorName),
+        (*env)->CallStaticObjectMethod(
+            env, jColorClass, jColorRgbMethod,
+            (int)([c redComponent] * 255.0f),
+            (int)([c greenComponent] * 255.0f),
+            (int)([c blueComponent] * 255.0f),
+            (double)[c alphaComponent]));
+}
+
++ (void)putColors:(jobject)preferences key:(const char*)colorName value:(NSArray*)colors {
+    GET_MAIN_JENV;
+
+    int count = [colors count];
+    jobjectArray res = (*env)->NewObjectArray(env, count, jColorClass, nil);
+
+    for (int i = 0; i < count; ++i) {
+        NSColor* c = [colors[i] colorUsingColorSpace:[NSColorSpace deviceRGBColorSpace]];
+        jobject fxcolor = (*env)->CallStaticObjectMethod(
+            env, jColorClass, jColorRgbMethod,
+            (int)([c redComponent] * 255.0f),
+            (int)([c greenComponent] * 255.0f),
+            (int)([c blueComponent] * 255.0f),
+            (double)[c alphaComponent]);
+
+        (*env)->SetObjectArrayElement(env, res, i, fxcolor);
+    }
+
+    (*env)->CallObjectMethod(env, preferences, jMapPutMethod, (*env)->NewStringUTF(env, colorName), res);
```

Comment:
Unless this is returning directly to Java (i.e., is called from a JNI method that will return right away), you should check for pending exceptions / NULL value.